provider.go

```diff
@@ -85,11 +108,41 @@ func (h *Handler) ListKeys(ctx context.Context, bucket string, cursor *uint64) (
 // Implementation of wasi:keyvalue/atomics
 func (h *Handler) Increment(ctx context.Context, bucket string, key string, delta uint64) (*wrpc.Result[uint64, atomics.Error], error) {
 	h.Logger.Debug("received request to increment key by delta", "key", key, "delta", delta)
-	res, err := h.collection.Binary().Increment(key, &gocb.IncrementOptions{Initial: int64(delta), Delta: delta})
+	collection, err := h.getCollectionFromContext(ctx)
+	res, err := collection.Binary().Increment(key, &gocb.IncrementOptions{Initial: int64(delta), Delta: delta})
 	if err != nil {
 		h.Logger.Error("unable to increment value at key", "key", key, "error", err)
 		return wrpc.Err[uint64](*errInvalidDataType), err
 	}
 
 	return wrpc.Ok[atomics.Error](res.Content()), nil
 }
+
+func (h *Handler) updateCouchbaseCluster(handler *Handler, sourceId string, config map[string]string) {
+	// Connect to the cluster
+	cluster, err := gocb.Connect(config["connectionString"], gocb.ClusterOptions{
```

Comment:
For these config accesses, in a future improvement, we should add some safety checks to make sure we don't panic on invalid config

Reply:
i agree . will work on that